### ISSUE TITLE
'Delete' button is not accessible using screen reader.

### ISSUE THREAD

**Issue by msftedad** [nvaccess/nvda#13755]
### Steps to reproduce:

Pre-requisite:
1.Open URL: https://www.bing.com/webmasters on New Edge Browser.
2.Login using valid credentials. Navigate to Add site manually section.
3.Add any site manually. Home page appears.

Repro Steps:
1. Open URL https://www.bing.com/webmasters on New Edge Browser.
2. Login using valid credentials.
3. Navigate to 'Settings' appearing in Header controls and activate it.
4. Popup appears.
5. Navigate to  API Access and activate it.
5. Navigate to OAuth Client and activate it.
7. O Auth client screen appears.
8. Open NVDA.
9. Navigate to 'Delete' button and activate it.
10.Observe an issue that Button 'Delete' is not accessible using screen reader. After activating, 'O Auth name' list gets expanded.

### Actual behavior:
'Delete' button is not getting activate with the screen reader. After activating, 'O Auth name' list gets expanded.

### Expected behavior:
After activating the 'delete' control, delete popup should appear. Control(s) must be accessible to Bluetooth keyboards and other assistive technology. Common causes of this problem include a) the element does not have a proper role assigned, b) the element needs tab-index="0" attribute to be focusable, or c) the component is not registering keypresses.

**SME Recommendation**
Fix the nesting of HTML elements so that the Delete button is not nested within another button. This invalid nesting is what is causing issues when trying to activate the Delete button with the screen reader.

#### NVDA installed/portable/running from source:
OS: Windows_11
Version: 21H2
OS Build: 22000.194
Browser: New edge (94.0.992.38)
URL:  https://www.bing.com/webmasters
Screen Reader: NVDA(2020.3)

#### NVDA version:
NVDA(2020.3)

#### Windows version:
OS: Windows_11
Version: 21H2
OS Build: 22000.194

**Screenshot**
<img width="608" alt="image" src="https://user-images.githubusercontent.com/93735775/171135569-d688fb21-994d-46eb-9b8b-ef1a91cc3417.png">

**Comment by XLTechie** [comment]
Can you test this with Narrator, and determine if the delete button is available 
to the user via that screen reader?

**Comment by seanbudd** [comment]
We are unable to process this issue as it stands.

As mentioned by the SME, this is a problem with the implementation of the application/website, as a starting point please submit feedback to the authors to let them know about the problem.

Kind Regards,
NV Access Software Developers
